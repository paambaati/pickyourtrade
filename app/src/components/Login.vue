<template>
  <div class="container">
    <section class="section">
      <div class="columns is-centered">
          <div class="column is-5-tablet is-4-desktop is-4-widescreen">
            <b-loading :active.sync="isLoggingIn" :is-full-page="false" :can-cancel="false">
            </b-loading>
            <form role="form" class="box" v-on:submit.prevent="login()">
              <b-message :active.sync="loginFailed" type="is-danger" aria-close-label="Close alert">
                ðŸ˜²Login failed. Please check your credentials!
              </b-message>
              
              <b-field label="Username">
                  <b-input v-model="username" required>
                  </b-input>
              </b-field>

              <b-field label="Password">
                  <b-input v-model="password" type="password" password-reveal required>
                  </b-input>
              </b-field>
              <b-button native-type="submit" class="is-primary">Login</b-button>
            </form>
          </div>
      </div>
    </section>
  </div>
</template>

<script>
export default {
  name: 'login',
  data() {
    return {
      username: '',
      password: '',
      isLoggingIn: false,
      loginFailed: false,
    };
  },
  methods: {
    login: function() {
      this.isLoggingIn = true;
      this.loginFailed = false;
      const username = this.username;
      const password = this.password;
      this.$store
        .dispatch('login', { username, password })
        .then(() => this.$router.push('/tracker'))
        .catch(() => {
          this.loginFailed = true;
        })
        .finally(() => {
          this.isLoggingIn = false;
        });
    }
  }
};
</script>
