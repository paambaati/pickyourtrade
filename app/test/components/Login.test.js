import { shallowMount, createLocalVue } from '@vue/test-utils';
import Login from '../../src/components/Login';
import Vuex from 'vuex';

// Component tests.
// This is to illustrate testing Vue components.
describe('âœ¨Login.test.js', () => {
    const localVue = createLocalVue();
    let component;

    beforeEach(() => {
        component = shallowMount(Login, {
            localVue,
            data() {
                return {
                    username: 'test1',
                    password: 'pass1',
                    isLoggingIn: false,
                    loginFailed: false,
                };
            },
        });
    });

    test('renders correctly', () => {
        const wrapper = shallowMount(Login);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('sets data correctly', () => {
        expect(component.vm.username).toEqual('test1');
        expect(component.vm.password).toEqual('pass1');
        expect(component.vm.isLoggingIn).toEqual(false);
        expect(component.vm.loginFailed).toEqual(false);
    });

    test('submitting the login form triggers the login action', () => {
        const loginMock = jest.fn(() => Promise.resolve());
        localVue.use(Vuex);
        const store = new Vuex.Store({
            actions: {
                login: loginMock,
            },
        });
        const wrapper = shallowMount(Login, { localVue, store });
        wrapper.find('form').trigger('submit');
        expect(loginMock).toHaveBeenCalled();
    });
});
