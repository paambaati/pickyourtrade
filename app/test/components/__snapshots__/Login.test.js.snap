// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`âœ¨Login.test.js renders correctly 1`] = `
<div
  class="container"
>
  <section
    class="section"
  >
    <div
      class="columns is-centered"
    >
      <div
        class="column is-5-tablet is-4-desktop is-4-widescreen"
      >
        <b-loading-stub
          animation="fade"
          oncancel="function _default() {}"
        />
         
        <form
          class="box"
          role="form"
        >
          <b-message-stub
            ariacloselabel="Close alert"
            closable="true"
            duration="2000"
            type="is-danger"
          >
            
              ðŸ˜²Login failed. Please check your credentials!
            
          </b-message-stub>
           
          <b-field-stub
            addons="true"
            label="Username"
          >
            <b-input-stub
              customclass=""
              hascounter="true"
              required=""
              type="text"
              usehtml5validation="true"
              value=""
            />
          </b-field-stub>
           
          <b-field-stub
            addons="true"
            label="Password"
          >
            <b-input-stub
              customclass=""
              hascounter="true"
              passwordreveal="true"
              required=""
              type="password"
              usehtml5validation="true"
              value=""
            />
          </b-field-stub>
           
          <b-button-stub
            class="is-primary"
            nativetype="submit"
            tag="button"
          >
            Login
          </b-button-stub>
        </form>
      </div>
    </div>
  </section>
</div>
`;
